spec/queries/keywords_query_spec.rb

```diff
@@ -38,6 +38,30 @@
           expect(keywords.count).to eq 2
         end
       end
+
+      context 'given the user search for a word vpn and match at least 1' do
+        it 'test' do
+          user = Fabricate(:user)
+          result_urls = ['https://www.thetopvpn.com', 'https://www.nordvpn.com', 'https://www.vnexpress.net']
+          keyword = Fabricate(:keyword, result_urls: result_urls, user: user)
+          filter_params = { word: 'vpn', match_at_least: '1' }
+          keywords = described_class.new(Keyword, filter_params).call
+
+          expect(keywords).to eq [keyword]
+        end
+      end
+
+      context 'given the user search for vpn word e and match at least 2' do
```

Comment:
Now that we gives priority to `adwords_url_contains` over `match_at_least`, I think we should add spec with context 'given the user search for adwords, word and match at least`.

Also, on that note, we should test with `word` without `match at least` and `match at a least` without `word` too 🙏 .

Reply:
solved in 33a43db

Reply:
How about this one to test priority declared in `keyword_query` 😄 
```
          filter_params = { adwords_url_contains: 'word' word: 'vpn', match_at_least: '2' }
# should return adwords result
```